### DISCUSSION TITLE
Discuss the loading pattern

### DISCUSSION THREAD

**Discussion by johanstromqvist** [Shopify/polaris-react#5976] [Pattern documentation]
Hey – thanks for showing interest in the loading guidelines. We'd love your feedback and ideas on improvements.

Find the guidelines here: https://polaris.shopify.com/patterns/loading

**Planned improvements**
- [ ] Add visual examples to clarify recommendation
- [ ] Separate best practices and patterns, e.g. "Minimize visual noise" vs "Use spinner when..."

**Comment by peterlazzarino** [comment]
Thanks for putting this together, really top of mind right now with the work to migrate plus features to admin.

For **Make it fast**, it would be great if we could either in this doc, or elsewhere define best practices and example approaches for caching, prefetching and managing page content, as well as potential techniques for loading on scroll since we'd probably want consistency with that experience.

also +10000 to **Make the layout stable**, if I had a dollar every time I click an element just as another loads over it async 😠 

I'm curious about - **Give clear feedback on the interaction** - is there an opinion around showing just that the state has changed, or should we go further to re-state what is happening such as a banner that says.. "saving..." to reinforce the action by the user.
